cmd/install/assets/hypershift_operator.go

```diff
@@ -103,6 +104,141 @@ func (o HyperShiftOperatorOIDCProviderS3Secret) Build() *corev1.Secret {
 	return secret
 }
 
+type ExternalDNSCredsSecret struct {
+	Namespace  *corev1.Namespace
+	CredsBytes []byte
+}
+
+func (o ExternalDNSCredsSecret) Build() *corev1.Secret {
+	secret := &corev1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Secret",
+			APIVersion: corev1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      externaDNSCredsSecretName,
+			Namespace: o.Namespace.Name,
+		},
+		Data: map[string][]byte{
+			"credentials": o.CredsBytes,
+		},
+	}
+	return secret
+}
+
+type ExternalDNSDeployment struct {
+	Namespace         *corev1.Namespace
+	Image             string
+	ServiceAccount    *corev1.ServiceAccount
+	Provider          string
+	DomainFilter      string
+	CredentialsSecret *corev1.Secret
+}
+
+func (o ExternalDNSDeployment) Build() *appsv1.Deployment {
+	replicas := int32(1)
+	deployment := &appsv1.Deployment{
```

Comment:
to reduce the number of moving parts, have we considered this to be a container of the HyperShift operator deployment?

Reply:
I had not considered it.  What are the trade-offs?

I personally wouldn't consider that to be less complex.  I would have to `-c` whenever I look at the pod logs for the HO.

Reply:
Also from a dev pov If I run the `hypershift-operator` binary in my machine I'd expect it to manage any internal component it needs for a HostedCluster to succeed?

Reply:
@csrwng and I talked about this arrived at the conclusion that we don't want to pull `external-dns` code into the `hypershift-operator` and we would just document the requirement for `external-dns` if you want DNS indirection.  As always, the CLI is helping with this for developer experience and CI tooling.